components/Button.tsx

```diff
@@ -0,0 +1,37 @@
+import { css, SerializedStyles } from '@emotion/react';
+
+const styles = {
+  button: css({
+    padding: '0.5em 2em'
+  })
+};
+
+type ButtonProps = {
+  color?: string;
+  backgroundColor?: string;
+  children?: React.ReactNode;
+  stylesProp?: SerializedStyles;
+} & React.ComponentProps<'button'>;
+
+const Button = ({
+  color = '#000',
+  backgroundColor = '#91E220',
+  stylesProp,
+  children,
+  ...rest
+}: ButtonProps) => {
+  return (
+    <button
+      css={[
+        styles.button,
+        { color: `${color}`, backgroundColor: `${backgroundColor}` },
```

Comment:
If you do want to pass styles generally through and avoid bloating the prop interface everytime you'd like to pass through something, it may be worthwhile to see how to pass styles through (the output type of the `css` function`)

Reply:
Sorry what do you mean by 'pass styles through'? Do you mean I wouldn't have to set color and backgroundColor manually the way that I'm doing?

Reply:
Indeed.

Something like this may be good `<CustomComponent css={[styles.x, etc, etc]} />`

and 

```
const styles = {..}

const CustomComponent = ({css}) => {

  return <div css={[...styles, ...css]} />
}
```

Reply:
Ohh okay. It makes sense. I literally just used that pattern for setting the button type as well. I should've clocked on and done that for CSS. Thanks.